src/dotnet/commands/dotnet-migrate/MigrateCommand.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using Microsoft.Build.Construction;
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.ProjectJsonMigration;
+
+namespace Microsoft.DotNet.Tools.Migrate
+{
+    public partial class MigrateCommand
+    {
+        private string _templateFile; 
+        private string _outputDirectory;
+        private string _projectJson;
+        private string _sdkVersion;
+
+        private TemporaryDotnetNewTemplateProject _temporaryDotnetNewProject;
+
+        public MigrateCommand(string templateFile, string outputDirectory, string projectJson, string sdkVersion)
+        {
+            _templateFile = templateFile;
+            _outputDirectory = outputDirectory;
+            _projectJson = projectJson;
+            _sdkVersion = sdkVersion;
+
+            _temporaryDotnetNewProject = new TemporaryDotnetNewTemplateProject();
+        }
+
+        public int Start()
+        {
+            var project = GetProjectJsonPath(_projectJson) ?? _temporaryDotnetNewProject.ProjectJsonPath;
```

Comment:
I'm not sure I follow this logic.  If `_projectJson` is null, this will use the "temporary" project.json file that was created with `dotnet new -t msbuild`?

I thought if `_projectJson` is null, it should use the project.json in the current directory.

```
app.Option("-p|--project", "The path to the project to run (defaults to the current directory). Can be a path to a project.json or a project directory", CommandOptionType.SingleValue);
```